app/js/services/services.js

```diff
@@ -534,6 +534,37 @@ KMCServices.factory('PlayerService', ['$http', '$modal', '$log', '$q', 'apiServi
 				);
 				return deferred.promise;
 			},
+			'playerUpgradeV7': function(playerObj, mode, templateId) {
+				var deferred = $q.defer();
+				var paramsPromise;
+				switch (mode) {
+					case 'default':
+						paramsPromise = PlayerDataService.getDefaultV7PlayerConf(playerObj.name, playerObj.description);
+						break;
+					case 'template':
+						paramsPromise = PlayerDataService.getV7PlayerConfFromTemplate(playerObj.name, templateId);
+						break;
+					default:
+						deferred.reject('Unknown upgrade mode "' + mode + '"');
+				}
+
+				paramsPromise.then(function (request) {
+					request['service'] = 'uiConf';
+					request['action'] = 'update';
+					request['id'] = playerObj.id;
+
+					var rejectText = $filter('translate')('Upgrade player action was rejected: ');
+
+					apiService.doRequest(request).then(function (result) {
+							deferred.resolve(result);
+						}, function (msg) {
+							deferred.reject(rejectText + msg);
+						}
+					);
+				});
+
+				return deferred.promise;
```

Comment:
please split the logic into individual functions where each has a single responsibility
```suggestion
var V7ConfPromise = function getV7Conf(mode, playerObj, templateId = null) { ... }
if(!V7ConfPromise){
  return deferred.reject('Unknown upgrade mode "' + mode + '"');
}
...
```